### PR TITLE
Update A8 payload

### LINKED ISSUE #1335: Assertion 8 needs to be more generic
payload should be: `networks`(empty, polkadot, ethereum,etc)

assertion: check `User has over X number of transactions` on the given networks. If `networks` is empty, assertion will check on all supported networks.

### PR DESCRIPTION
**Pull request by zTgx**
resolved #1335 

* update A8 payload
* rename issuer's field `shard` to `mrenclave`
* update `tee-worker/ts-test`

### CODE REVIEW

File: tee-worker/litentry/core/assertion-build/src/a8.rs [hunk 1]
```diff
@@ -23,38 +23,175 @@ extern crate sgx_tstd as std;
...
+	};
+}
+
+fn assertion_networks_to_vc_networks(
+	networks: &AssertionNetworks,
+) -> HashSet<VerifiedCredentialsNetwork> {
+	let mut set: HashSet<VerifiedCredentialsNetwork> = HashSet::new();
+
+	if networks.is_empty() {
+		NETWORK_HASHSET.clone()
```
Comment (jingleizhang) [inline tee-worker/litentry/core/assertion-build/src/a8.rs:67]: If the input is empty, the network vector should be empty as well, to keep the computation at a minimum level.

File: tee-worker/litentry/core/credentials/src/templates/a8.json [hunk 2]
```diff
@@ -23,19 +23,30 @@
...
-                    "dst":""
-                },
-                {
-                    "src":"$total_txs",
-                    "op":"<=",
-                    "dst":""
-                }
-            ]
-        }],
+			"and": [{
```
Comment (jingleizhang) [inline tee-worker/litentry/core/credentials/src/templates/a8.json]: please format those plain texts

File: tee-worker/litentry/core/assertion-build/src/a8.rs [hunk 3]
```diff
@@ -23,38 +23,175 @@ extern crate sgx_tstd as std;
...
+								set.insert(khala);
+							},
+							"ethereum" => {
+								let ethereum =
+									VerifiedCredentialsNetwork::from(EvmNetwork::Ethereum);
+								set.insert(ethereum);
+							},
+							_ => {
+								info!("		[AssertionBuild-A8] Wrong Network!");
+							},
```
Comment (DogLi) [inline tee-worker/litentry/core/assertion-build/src/a8.rs:112]: as `SubstrateNetwork` already derive `Serialize/Deserialize`, maybe we can use [serde_plain](https://crates.io/crates/serde_plain) to simplify this code:

```
impl FromStr for SubstrateNetwork {
    fn from_str() -> Result<String, xxx> {
        // use serde_plain here
    } 
}
```

File: tee-worker/litentry/core/assertion-build/src/a8.rs [hunk 4]
```diff
@@ -23,38 +23,175 @@ extern crate sgx_tstd as std;
...
+			},
+			VerifiedCredentialsNetwork::Kusama => {
+				rets.push("kusama");
+			},
+			VerifiedCredentialsNetwork::Khala => {
+				rets.push("khala");
+			},
+			VerifiedCredentialsNetwork::Ethereum => {
+				rets.push("ethereum");
+			},
```
Comment (DogLi) [inline tee-worker/litentry/core/assertion-build/src/a8.rs:151]: impl `ToString` for the `VerifiedCredentialsNetwork ` by using the `serde_plain`

**Review by jingleizhang — APPROVED** [review]
LGTM, thanks!

File: tee-worker/litentry/core/assertion-build/src/a8.rs (hunk 1 above)
Comment (jingleizhang) [reply tee-worker/litentry/core/assertion-build/src/a8.rs:67]: As the issue mentioned (https://github.com/litentry/litentry-parachain/issues/1335#issue-1587128272): If `network` is empty, the assertion will check on all supported networks. 

So this is good.

**Pull request merged**